icu4j/main/classes/core/src/com/ibm/icu/impl/units/ComplexUnitsConverter.java

```diff
@@ -129,15 +129,17 @@ public boolean greaterThanOrEqual(BigDecimal quantity, BigDecimal limit) {
                 // decision is made. However after the thresholding, we use the
                 // original values to ensure unbiased accuracy (to the extent of
                 // double's capabilities).
-                BigDecimal roundedQuantity =
+                BigDecimal flooredQuantity =
                     quantity.multiply(EPSILON_MULTIPLIER).setScale(0, RoundingMode.FLOOR);
-                intValues.add(roundedQuantity);
+                intValues.add(flooredQuantity);
 
                 // Keep the residual of the quantity.
                 //   For example: `3.6 feet`, keep only `0.6 feet`
-                quantity = quantity.subtract(roundedQuantity);
-                if (quantity.compareTo(BigDecimal.ZERO) == -1) {
+                BigDecimal remainder = quantity.subtract(flooredQuantity);
```

Comment:
Looking at this again, I do notice I'm adding a new variable without much reason - I'd be happy to change this back to just using quantity directly if someone prefers?
(`quantity = quantity.subtract(flooredQuantity);` for J, and `quantity -= flooredQuantity;` for C.)